dj/models/node.py

```diff
@@ -82,15 +110,25 @@ class NodeYAML(TypedDict, total=False):
     tables: Dict[str, List[TableYAML]]
 
 
-class Node(SQLModel, table=True):  # type: ignore
+class NodeBase(SQLModel):
     """
-    A node.
+    A base node.
     """
 
-    id: Optional[int] = Field(default=None, primary_key=True)
     name: str = Field(sa_column=SqlaColumn("name", String, unique=True))
     description: str = ""
+    type: NodeType = Field(sa_column=SqlaColumn(Enum(NodeType)))
+    query: Optional[str] = None
+    mode: NodeMode = NodeMode.PUBLISHED
```

Comment:
Shouldn't default mode be `DRAFT`?

Reply:
This depends on the behavior we're expecting the API to have.

When someone makes a request to create a node, do we always assume it's to be published right away? Published requires that `status="valid"` and in the case where the node is invalid we would return an error with additional text along the lines of `set mode to "draft" if you'd like to create an invalid node`.

On the other hand if we default to draft mode, then the common flow is that users face little friction when adding a node, assuming that the simple checks pass (name is not already in use, SQL is syntactically valid, etc.), and then users are familiar that they have to publish the node at some point which is when they'd face the requirement of the node being valid.

Reply:
Yeah... I think this will be clearer as we go. Probably OK to leave the stringer default for starters.